cosmos/x/evm/plugins/precompile/plugin.go

```diff
@@ -164,7 +163,14 @@ func (p *plugin) Run(
 // EnableReentrancy sets the state so that execution can enter the EVM again.
 //
 // EnableReentrancy implements core.PrecompilePlugin.
-func (p *plugin) EnableReentrancy(_ context.Context) {
+func (p *plugin) EnableReentrancy(evm ethprecompile.EVM) {
+	sdb := utils.MustGetAs[vm.PolarisStateDB](evm.GetStateDB())
```

Comment:
Can we move the re-entrancy guard to the `eth` folder?

Reply:
this is logic specific to Cosmos

Reply:
that we need to stop/start emitting logs, switch gas config

Reply:
Check slack my bad